edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/RoutingEdgeHub.cs

```diff
@@ -156,11 +156,7 @@ public async Task AddSubscription(string id, DeviceSubscription deviceSubscripti
             }
             catch (Exception e)
             {
-                Events.ErrorAddingSubscription(e, id, deviceSubscription);
-                if (!e.HasTimeoutException())
-                {
-                    throw;
-                }
+                Events.ErrorAddingSubscription(e, id, deviceSubscription);                
```

Comment:
When does this subscription finally get cleaned up?

Reply:
We store the subscription, and on Connection changed event, we re-process the subscriptions for the client.